engine.go

```diff
@@ -163,6 +164,23 @@ func SubworkflowCache(wf *workflow.Workflow, rootDir string) (loadfile.FileCache
 	if err != nil {
 		return nil, err
 	}
+	for _, ctxFile := range subworkflowCache.Files() {
+		subwf, err := converter.FromYAML(ctxFile.Content)
+		if err != nil {
+			return nil, err
+		}
+		flowCache, err := SubworkflowCache(subwf, rootDir, converter, flowCaches)
+		if err != nil {
+			return nil, err
+		}
+		flowCaches = append(flowCaches, flowCache)
+	}
+
+	flowCaches = append(flowCaches, subworkflowCache)
+	subworkflowCache, err = loadfile.MergeFileCaches(flowCaches...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
So you're recursively creating a file cache for each subworkflow YAML, and then merging them?

As long as all the fully resolved file paths are unique (and we're always currently working from the same root anyway), why not just put everything into one single file cache object? If there are any conflicting keys, they have to represent the same actual file.

We talked before about "stacking" or "chaining" file caches, but that was particularly targeted at avoiding changing the root used for previously loaded files, and it doesn't really apply if the root can't change.

I don't see any obvious reason that all the subworkflows can't "just get along" (in the same cache map) as the code is currently structured.

Am I missing something?